### ISSUE TITLE
TypeError: Cannot read property '0' of undefined

### ISSUE THREAD

**Issue by adrienharnay** [typescript-eslint/typescript-eslint#270]
**Repro**

```JS
    {
      files: ['*.ts?(x)'],
      parser: '@typescript-eslint/parser',
      plugins: ['@typescript-eslint'],
      rules: {
        'react/prop-types': 'off',
        '@typescript-eslint/adjacent-overload-signatures': 'error',
        '@typescript-eslint/array-type': ['error', 'array'],
        '@typescript-eslint/generic-type-naming': ['error', '^[a-zA-Z]+$'],
        '@typescript-eslint/no-angle-bracket-type-assertion': 'error',
        '@typescript-eslint/no-array-constructor': 'error',
        '@typescript-eslint/no-empty-interface': 'error',
        '@typescript-eslint/no-explicit-any': 'error',
        '@typescript-eslint/no-extraneous-class': 'error',
        '@typescript-eslint/no-inferrable-types': 'error',
        '@typescript-eslint/no-misused-new': 'error',
        '@typescript-eslint/no-namespace': 'error',
        '@typescript-eslint/no-non-null-assertion': 'error',
        '@typescript-eslint/no-object-literal-type-assertion': 'error',
        '@typescript-eslint/no-parameter-properties': 'error',
        '@typescript-eslint/no-this-alias': 'error',
        '@typescript-eslint/no-triple-slash-reference': 'error',
        '@typescript-eslint/no-type-alias': [
          'error',
          {
            allowAliases: 'always',
            allowCallbacks: 'always',
            allowMappedTypes: 'always',
          },
        ],
        '@typescript-eslint/no-unused-vars': [
          'error',
          { ignoreRestSiblings: true },
        ],
        '@typescript-eslint/prefer-interface': 'error',
        '@typescript-eslint/prefer-namespace-keyword': 'error',
        '@typescript-eslint/type-annotation-spacing': 'error',
      },
    },
```

```TS
export type ButtonProps = JSX.IntrinsicElements['button'];
```

**Expected Result**

No crash

**Actual Result**

```
TypeError: Cannot read property '0' of undefined
    at Object.exports.upperCaseFirst.str [as upperCaseFirst] (/Path/node_modules/@typescript-eslint/eslint-plugin/lib/util.js:107:36)
    at getMessage (/Path/node_modules/@typescript-eslint/eslint-plugin/lib/rules/no-type-alias.js:209:26)
    at validateTypeAliases (/Path/node_modules/@typescript-eslint/eslint-plugin/lib/rules/no-type-alias.js:258:24)
    at node.types.forEach.type (/Path/node_modules/@typescript-eslint/eslint-plugin/lib/rules/no-type-alias.js:273:11)
    at Array.forEach (<anonymous>)
    at validateCompositions (/Path/node_modules/@typescript-eslint/eslint-plugin/lib/rules/no-type-alias.js:269:18)
    at validateNode (/Path/node_modules/@typescript-eslint/eslint-plugin/lib/rules/no-type-alias.js:287:9)
    at TSTypeAliasDeclaration (/Path/node_modules/@typescript-eslint/eslint-plugin/lib/rules/no-type-alias.js:298:9)
    at listeners.(anonymous function).forEach.listener (/Path/node_modules/eslint/lib/util/safe-emitter.js:45:58)
    at Array.forEach (<anonymous>)
    at Object.emit (/Path/node_modules/eslint/lib/util/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/Path/node_modules/eslint/lib/util/node-event-generator.js:251:26)
    at NodeEventGenerator.applySelectors (/Path/node_modules/eslint/lib/util/node-event-generator.js:280:22)
    at NodeEventGenerator.enterNode (/Path/node_modules/eslint/lib/util/node-event-generator.js:294:14)
    at CodePathAnalyzer.enterNode (/Path/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:632:23)
    at nodeQueue.forEach.traversalInfo (/Path/node_modules/eslint/lib/linter.js:750:28)
```

**Additional Info**

```TS
export type ButtonProps = PropsWithoutRef<JSX.IntrinsicElements['button']>;
```

does work

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.3.0` |
| `@typescript-eslint/parser`        | `1.3.0` |
| `TypeScript`                       | `3.3.3` |
| `ESLint`                           | `5.13.0` |
| `node`                             | `10.15.0` |
| `npm`                              | `6.4.1` |
| `yarn`                             | `1.13.0` |

**Comment by armano2** [comment]
looks like issue is no longer present in master and is fixed in 1.3.1-alpha13

**Comment by adrienharnay** [comment]
Oh sweet, will test it after the next release :+1:
